### PR TITLE
Port of existing KinesisTransport

### LINKED ISSUE #78: Kinesis Transport part 2: Port existing AWS Kinesis transport over to the new AWS Input
The existing KinesisTransport works well we spent a lot of time fine tuning it in the past year. So, the first step in implementing a Kinesis Transport for the new AWS integrations is to port the existing one over and get it working. It uses the old 1.x version of the Kinesis Client Library, but that should be ok to start with. 

To complete this issue, just copy over all KinesisTransport classes and components, and get to the point where you can start the Graylog server without any syntax errors.

See this class to get started. This class references several other classes that will need to be copied over too.
https://github.com/Graylog2/graylog-plugin-aws/blob/2bc458ed5a5d6a55a9014c6edb72f48aeb2534dc/src/main/java/org/graylog/aws/inputs/transports/KinesisTransport.java

### PR DESCRIPTION
**Pull request by danotorrey**
## Overview
Port of the existing KinesisTransport to the new AWS integration. This is a direct port of `KinesisTransport` and `KinesisConsumer` classes. This was the fastest way to get this integration working. This existing KinesisTransport uses the Kinesis Client Library v1. The runtime logic works great (and supports throttling), however it has poor state management (fails to shut down and restart properly). I did not attempt to fix this issue (see Next Steps below).

The only changes were in the message processing area right before the message is queued to be written to the journal. Now, the [KinesisPayloadDecoder](https://github.com/Graylog2/graylog-plugin-integrations/blob/bf2ffc416c98bff82889cf932104bd9fe050366d/src/main/java/org/graylog/integrations/aws/transports/KinesisPayloadDecoder.java) is used to handle the pre-journal message processing (decompress and unpack CloudWatch messages from GZipped JSON payload so that individual messages can be written to the journal).

https://github.com/Graylog2/graylog-plugin-integrations/blob/bf2ffc416c98bff82889cf932104bd9fe050366d/src/main/java/org/graylog/integrations/aws/transports/KinesisConsumer.java#L169-L172

Closes #78

### CODE REVIEW

**Comment by bernd** [comment]
@danotorrey Can you please fix the conflicts? :smile: Thanks!

**Comment by danotorrey** [comment]
> @danotorrey Can you please fix the conflicts? 😄 Thanks!

@bernd Fixing now...

**Comment by danotorrey** [comment]
@bernd Merge conflicts fixed.

File: src/main/java/org/graylog/integrations/aws/transports/AWSTransport.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import com.google.common.eventbus.EventBus;
 import com.google.inject.assistedinject.Assisted;
 import org.graylog.integrations.aws.AWSMessageType;
+import org.graylog.integrations.aws.codecs.AWSCodec;
 import org.graylog.integrations.aws.inputs.AWSInput;
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/transports/AWSTransport.java]: import is not being used

File: src/test/java/org/graylog/integrations/aws/codecs/AWSCodecTest.java [hunk 2]
```diff
@@ -31,7 +31,7 @@ public void setUp() throws Exception {
     public void testKinesisFlowLogCodec() throws JsonProcessingException {
 
         final HashMap<String, Object> configMap = new HashMap<>();
-        configMap.put(AWSInput.CK_AWS_MESSAGE_TYPE, AWSMessageType.KINESIS_FLOW_LOGS.toString());
```
Comment (ceruleancee) [inline src/test/java/org/graylog/integrations/aws/codecs/AWSCodecTest.java]: AWSInput import class is no longer needed

File: src/test/java/org/graylog/integrations/aws/transports/AWSTransportTest.java [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 import com.google.common.eventbus.EventBus;
 import org.graylog.integrations.aws.AWSMessageType;
+import org.graylog.integrations.aws.codecs.AWSCodec;
 import org.graylog.integrations.aws.inputs.AWSInput;
```
Comment (ceruleancee) [inline src/test/java/org/graylog/integrations/aws/transports/AWSTransportTest.java]: import not needed

**Comment by danotorrey** [comment]
I am working on upgrading the KCL to the latest version. See #151. The upgrade works and processes, and it successfully processes messages.

Now, I am going to investigate if I can resolve the state management (start/restart) issues.

**Pull request merged**